userspace/falco/stats_writer.cpp

```diff
@@ -625,7 +629,7 @@ void stats_writer::collector::collect(const std::shared_ptr<sinsp>& inspector,
 				flags &= ~(METRICS_V2_KERNEL_COUNTERS | METRICS_V2_KERNEL_COUNTERS_PER_CPU |
 				           METRICS_V2_STATE_COUNTERS | METRICS_V2_LIBBPF_STATS);
 			}
-			m_writer->m_libs_metrics_collector =
+			m_writer->m_libs_metrics_collectors[src] =
```

Comment:
We cache it per-source.
In case of capture mode (ie: when we are reading a scap file), we will have multiple libs_metrics_collectors, one for each source, even if the inspector always stays the same (since all sources get attached to a single inspector in capture mode).